src/engine/controls/clockcontrol.cpp

```diff
@@ -30,27 +43,145 @@ void ClockControl::trackLoaded(TrackPointer pNewTrack) {
 
 void ClockControl::trackBeatsUpdated(mixxx::BeatsPointer pBeats) {
     // Clear on-beat control
-    m_pCOBeatActive->set(0.0);
+    m_pCOBeatActive->forceSet(0.0);
     m_pBeats = pBeats;
 }
 
 void ClockControl::process(const double dRate,
-                           const double currentSample,
-                           const int iBuffersize) {
+        const double currentSample,
+        const int iBuffersize) {
+    Q_UNUSED(dRate);
+    Q_UNUSED(currentSample);
     Q_UNUSED(iBuffersize);
-    double samplerate = m_pCOSampleRate->get();
+}
+
+void ClockControl::updateIndicators(const double dRate,
+        const double currentSample,
+        const double sampleRate) {
+    /* This method sets the control beat_active is set to the following values:
+    * +1.0 --> Forward playing, set at the beat and set back to 0.0 at 20% of beat distance
+    *  0.0 --> No beat indication (ouside 20% area or play direction changed while indication was on)
+    * -1.0 --> Reverse playing, set at the beat and set back to 0.0 at -20% of beat distance
```

Comment:
Did you consider to change this to two? This way we can directly bind it to a three state button with "<" and ">" or similar icons.

Reply:
Done!